include/SDL3/SDL_gpu.h

```diff
@@ -0,0 +1,2263 @@
+    /* Compressed SRGB Color Formats */
+    SDL_GPU_TEXTUREFORMAT_BC3_SRGB,
+    SDL_GPU_TEXTUREFORMAT_BC7_SRGB,
+    /* Depth Formats */
+    SDL_GPU_TEXTUREFORMAT_D16_UNORM,
+    SDL_GPU_TEXTUREFORMAT_D24_UNORM,
+    SDL_GPU_TEXTUREFORMAT_D32_SFLOAT,
+    SDL_GPU_TEXTUREFORMAT_D24_UNORM_S8_UINT,
+    SDL_GPU_TEXTUREFORMAT_D32_SFLOAT_S8_UINT
+} SDL_GpuTextureFormat;
+
+typedef enum SDL_GpuTextureUsageFlagBits
+{
+    SDL_GPU_TEXTUREUSAGE_SAMPLER_BIT = 0x00000001,
+    SDL_GPU_TEXTUREUSAGE_COLOR_TARGET_BIT = 0x00000002,
+    SDL_GPU_TEXTUREUSAGE_DEPTH_STENCIL_TARGET_BIT = 0x00000004,
+    SDL_GPU_TEXTUREUSAGE_GRAPHICS_STORAGE_READ_BIT = 0x00000008,
+    SDL_GPU_TEXTUREUSAGE_COMPUTE_STORAGE_READ_BIT = 0x00000020,
+    SDL_GPU_TEXTUREUSAGE_COMPUTE_STORAGE_WRITE_BIT = 0x00000040
+} SDL_GpuTextureUsageFlagBits;
+
+typedef Uint32 SDL_GpuTextureUsageFlags;
+
+typedef enum SDL_GpuTextureType
+{
+    SDL_GPU_TEXTURETYPE_2D,
+    SDL_GPU_TEXTURETYPE_3D,
+    SDL_GPU_TEXTURETYPE_CUBE,
+} SDL_GpuTextureType;
+
+typedef enum SDL_GpuSampleCount
+{
+    SDL_GPU_SAMPLECOUNT_1,
+    SDL_GPU_SAMPLECOUNT_2,
+    SDL_GPU_SAMPLECOUNT_4,
+    SDL_GPU_SAMPLECOUNT_8
+} SDL_GpuSampleCount;
+
+typedef enum SDL_GpuCubeMapFace
+{
+    SDL_GPU_CUBEMAPFACE_POSITIVEX,
+    SDL_GPU_CUBEMAPFACE_NEGATIVEX,
+    SDL_GPU_CUBEMAPFACE_POSITIVEY,
+    SDL_GPU_CUBEMAPFACE_NEGATIVEY,
+    SDL_GPU_CUBEMAPFACE_POSITIVEZ,
+    SDL_GPU_CUBEMAPFACE_NEGATIVEZ
+} SDL_GpuCubeMapFace;
+
+typedef enum SDL_GpuBufferUsageFlagBits
+{
+    SDL_GPU_BUFFERUSAGE_VERTEX_BIT = 0x00000001,
+    SDL_GPU_BUFFERUSAGE_INDEX_BIT = 0x00000002,
+    SDL_GPU_BUFFERUSAGE_INDIRECT_BIT = 0x00000004,
+    SDL_GPU_BUFFERUSAGE_GRAPHICS_STORAGE_READ_BIT = 0x00000008,
+    SDL_GPU_BUFFERUSAGE_COMPUTE_STORAGE_READ_BIT = 0x00000020,
+    SDL_GPU_BUFFERUSAGE_COMPUTE_STORAGE_WRITE_BIT = 0x00000040
+} SDL_GpuBufferUsageFlagBits;
+
+typedef Uint32 SDL_GpuBufferUsageFlags;
+
+typedef enum SDL_GpuTransferBufferUsage
+{
+    SDL_GPU_TRANSFERBUFFERUSAGE_UPLOAD,
+    SDL_GPU_TRANSFERBUFFERUSAGE_DOWNLOAD
+} SDL_GpuTransferBufferUsage;
+
+typedef enum SDL_GpuShaderStage
+{
+    SDL_GPU_SHADERSTAGE_VERTEX,
+    SDL_GPU_SHADERSTAGE_FRAGMENT
+} SDL_GpuShaderStage;
+
+typedef enum SDL_GpuShaderFormat
+{
+    SDL_GPU_SHADERFORMAT_INVALID,
+    SDL_GPU_SHADERFORMAT_SPIRV,    /* Vulkan */
+    SDL_GPU_SHADERFORMAT_HLSL,     /* D3D11, D3D12 */
+    SDL_GPU_SHADERFORMAT_DXBC,     /* D3D11, D3D12 */
+    SDL_GPU_SHADERFORMAT_DXIL,     /* D3D12 */
+    SDL_GPU_SHADERFORMAT_MSL,      /* Metal */
+    SDL_GPU_SHADERFORMAT_METALLIB, /* Metal */
+    SDL_GPU_SHADERFORMAT_SECRET    /* NDA'd platforms */
+} SDL_GpuShaderFormat;
+
+typedef enum SDL_GpuVertexElementFormat
+{
+    SDL_GPU_VERTEXELEMENTFORMAT_UINT,
+    SDL_GPU_VERTEXELEMENTFORMAT_FLOAT,
+    SDL_GPU_VERTEXELEMENTFORMAT_VECTOR2,
+    SDL_GPU_VERTEXELEMENTFORMAT_VECTOR3,
+    SDL_GPU_VERTEXELEMENTFORMAT_VECTOR4,
+    SDL_GPU_VERTEXELEMENTFORMAT_COLOR,
```

Comment:
I agree with an earlier comment that this COLOR enum is a confusing D3D-ism.

I think having an explicit enum of supported type+component combinations like this enum is better than having them separate and disallowing seemingly arbitrary combinations, though.

Reply:
Agreed, tracking at https://github.com/thatcosmonaut/SDL/issues/97